react/src/helper/big-number.ts

```diff
@@ -85,56 +68,78 @@ export namespace BigNumber {
     const bytes = (sizeValue as Big).times(base ** sizeIndex);
     let targetIndex = 0;
 
-    if (targetSizeUnit === 'auto') {
+    if (targetUnit === 'auto') {
       targetIndex = logBigNumber(bytes, base);
       targetIndex = Math.min(Math.max(targetIndex, 0), sizes.length - 1);
     } else {
-      targetIndex = targetSizeUnit
-        ? sizes.indexOf(targetSizeUnit.toUpperCase() as SizeUnit)
-        : sizeIndex;
+      targetIndex = targetUnit ? sizes.indexOf(targetUnit.toLowerCase()) : 0;
     }
 
     const finalBytes = bytes.div(base ** targetIndex);
     const numberFixed = round
       ? finalBytes.toFixed(fixed)
       : toFixedFloorWithoutTrailingZeros(finalBytes, fixed);
 
+    const unit = sizes[targetIndex] as InputSizeUnit;
+
     return {
       number: finalBytes,
       numberFixed,
-      unit: sizes[targetIndex],
-      numberUnit: `${numberFixed}${sizes[targetIndex]}`,
+      unit,
+      value: `${numberFixed}${unit}`,
     };
   }
 
-  export function convertBinarySizeUnit(
-    sizeWithUnit?: BigSource,
-    targetSizeUnit?: SizeUnit | 'auto',
+  export function convertToBinaryUnit(
+    inputValue?: BigSource,
+    targetUnit?: SizeUnit | 'auto',
     fixed = 2,
     round = false,
   ) {
-    return BigNumber.convertSizeUnit(
-      sizeWithUnit,
-      targetSizeUnit,
+    const convertedValue = BigNumber.convertUnitValue(inputValue, targetUnit, {
       fixed,
       round,
-      1024,
-    );
+      base: 1024,
+    });
+
+    if (convertedValue === undefined) {
```

Comment:
We can add sub-routine for display value to reduce duplications here and at 131 line

eg) return BigNumber.convertUnitValueToDisplayValue('iB' // or 'B');

Reply:
I pushed a change to refactor this part.